### DISCUSSION TITLE
Runwasi is suitable and enabled by default

### DISCUSSION THREAD

**Discussion by Jengro777** [moby/moby#47604] [Q&A] [answered]
```
	"features": {
		"containerd-snapshotter": true
	},
```
	
Currently, when using Docker for WASI runtime, it is necessary to enable configuration and download the corresponding runWASI runtime yourself.
1. Will this feature be enabled by default after it becomes stable
2. Will we automatically download the Wasmtime Wasmedge Wasmer runtime in the future?

**Comment by neersighted** [comment, marked as answer]
I see you are repeating your question from https://github.com/moby/moby/discussions/46550#discussioncomment-8859467, more-or-less.

To reiterate and clarify, of course the feature will be on by default when it is stable. We would like to do so in a major version or two, though the timeline is not definite and it's definitely a "when it's ready" change as it is high-risk and quite invasive. https://github.com/moby/moby/pull/47526 and follow-ups are very likely prerequisites as well.

Regarding Wasm runtimes/shims, that is not a moby/moby question. While we may add Wasm runtimes to the development environment/integration tests, pre-installing runtimes is a question for downstream packagers. For example, Docker CE packaging is at https://github.com/docker/docker-ce-packaging; the code there is what governs what is in packages from https://download.docker.com, and decisions to package (or not package) runtimes are made in that downstream project.
